lib/compress/zstd_lazy.c

```diff
@@ -644,20 +741,21 @@ ZSTD_compressBlock_lazy_generic(
     searchMax_f const searchMax = dictMode == ZSTD_dictMatchState ?
         (searchMethod==search_binaryTree ? ZSTD_BtFindBestMatch_dictMatchState_selectMLS
                                          : ZSTD_HcFindBestMatch_dictMatchState_selectMLS) :
+        (dictMode == ZSTD_dedicatedDictSearch ? ZSTD_HcFindBestMatch_dedicatedDictSearch_selectMLS :
         (searchMethod==search_binaryTree ? ZSTD_BtFindBestMatch_selectMLS
-                                         : ZSTD_HcFindBestMatch_selectMLS);
+                                         : ZSTD_HcFindBestMatch_selectMLS));
     U32 offset_1 = rep[0], offset_2 = rep[1], savedOffset=0;
 
     const ZSTD_matchState_t* const dms = ms->dictMatchState;
-    const U32 dictLowestIndex      = dictMode == ZSTD_dictMatchState ?
+    const U32 dictLowestIndex      = dictMode == ZSTD_dictMatchState || dictMode == ZSTD_dedicatedDictSearch ?
```

Comment:
nit: We should have a constant for this. If the compiler gets confused about propagating the constants it could be a macro instead.

Reply:
Parenthesis could help readability